### PR TITLE
Improve managed runtime detection

### PR DESCRIPTION
**Pull request by Tomasz-Smelcerz-SAP**
**Description**

This PR improves detection of managed Kyma runtimes in the CLI.
This is a temporary solution that addresses the current runtimes managed by the reconciler.
In the future, when we completely switch to the modularization-based solution, we'll have to adjust this logic.
See https://github.com/kyma-project/cli/issues/1472 for details. 

Changes proposed in this pull request:

- Add "marker" ConfigMap detection. This ConfigMap is only installed on SKR's
- Improve message handling (no more stack trace in the console)

**Related issue(s)**
- https://github.com/kyma-project/cli/issues/1472
- https://github.com/kyma-project/cli/issues/1460

### CODE REVIEW

File: cmd/kyma/undeploy/cmd.go
Comment (jakobmoellersap) [inline cmd/kyma/undeploy/cmd.go:87]:
```suggestion
		if err := cli.DetectManagedEnvironment(ctx, cmd.K8s, cmd.Factory.NewStep("")); err != nil {
```
you actually can use the command contact if you adjust the command Run above like this


```golang

	cobraCmd := &cobra.Command{
		Use:   "undeploy",
		Short: "Undeploys Kyma from a running Kubernetes cluster.",
		Long:  `Use this command to undeploy Kyma from a running Kubernetes cluster.`,
		RunE:  func(cobraCmd *cobra.Command, _ []string) error { return cmd.Run(cobraCmd.Context()) },
	}

```

and then pass it with

```golang
func (cmd *command) Run(ctx context.Context) error {
```

File: internal/clusterinfo/clusterinfo.go
Comment (jakobmoellersap) [inline internal/clusterinfo/clusterinfo.go:96]: this will be a huge list for some clusters and can lead to a very long lookup, is this accepted as tradeoff?

File: cmd/kyma/undeploy/cmd.go
Comment (Tomasz-Smelcerz-SAP) [reply cmd/kyma/undeploy/cmd.go:87]: :+1:

File: internal/clusterinfo/clusterinfo.go
Comment (Tomasz-Smelcerz-SAP) [reply internal/clusterinfo/clusterinfo.go]: Changed to `Get()` instead of `List()`

**Review by jakobmoellersap — COMMENTED** [review]
LGTM for me, just waiting for a second review of someone with more legacy reconciler background

File: internal/clusterinfo/clusterinfo.go
Comment (lindnerby) [inline internal/clusterinfo/clusterinfo.go:81]: ... and more :)

File: internal/clusterinfo/clusterinfo.go
Comment (lindnerby) [inline internal/clusterinfo/clusterinfo.go:83]: Noisy comment can be deleted I guess?

File: internal/clusterinfo/clusterinfo.go
Comment (lindnerby) [inline internal/clusterinfo/clusterinfo.go:103]:
```suggestion
		return false, fmt.Errorf("failed to get ConfigMap \"skr-configmap\"  in the \"kyma-system\" namespace: %w", err)
```

File: internal/cli/managed.go
Comment (lindnerby) [inline internal/cli/managed.go:23]:
```suggestion
			return errors.New("command stopped by user")
```

File: internal/clusterinfo/clusterinfo.go
Comment (lindnerby) [inline internal/clusterinfo/clusterinfo.go:85]: This comment also provides no informational value since the const ist already explaining legacy context. Can be removed?

**Review by lindnerby — COMMENTED** [review]
Just some minor clean code findings. At some point we should improve this detection mechanism though. Maybe back it up with information on the Kyma CR itself?

File: internal/clusterinfo/clusterinfo.go
Comment (Tomasz-Smelcerz-SAP) [reply internal/clusterinfo/clusterinfo.go:83]: +1

File: internal/clusterinfo/clusterinfo.go
Comment (Tomasz-Smelcerz-SAP) [reply internal/clusterinfo/clusterinfo.go:85]: :+1:

**Review by lindnerby — APPROVED** [review]

**Pull request merged**
